### DISCUSSION TITLE
Null discount disables sales

### DISCUSSION THREAD

**Discussion by paddling** [craftcms/commerce#2410] [Ideas]
**Description**
If a CP user creates and saves a discount BUT the discount settings are not populated with any details it will disable any active sales.

**To Reproduce**
1. In the CP, create an active sale in commerce.
2. Verify that sale is being applied as expected on the front-end.
1. In the CP, create a discount with all default values (just add a title) and save. Note the the default setting is to "ignore sales when this discount is applied to matching line items"   
![image](https://user-images.githubusercontent.com/58182286/92943552-e9fca180-f420-11ea-87e3-ca8c7c140a04.png)

1. On the front end, depending on display settings, the sale will still appear on the category page and product page but not in the cart.
1. Disable the new, null discount and the sale will work again.

**Expected behavior**
A null discount shouldn't be allowed. Some action should be required and/or the default value for "ignore sales when this discount is applied to matching line items" should NOT be checked.

**Additional info**
* Craft 3.5.8 
* Commerce 3.2.3

**Comment by lukeholder** [comment]
Thanks for the feedback and understand the concern. For now, you will just need to check the settings when saving a discount.

We have begun a refactor of the discount system for the next release, hopefully alleviating UX issues like this soon.

Keeping this ticket open until we do.
